File: .NET/Microsoft.Recognizers.Text.Sequence/English/Extractors/EnglishURLExtractorConfiguration.cs
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.

using System.Text.RegularExpressions;
using Microsoft.Recognizers.Definitions;

namespace Microsoft.Recognizers.Text.Sequence.English
{
    public class EnglishURLExtractorConfiguration : URLConfiguration
    {
        public EnglishURLExtractorConfiguration(SequenceOptions options)
            : base(options)
        {
            IpUrlRegex = new Regex(BaseURL.IpUrlRegex, RegexOptions.Compiled);
            UrlRegex = new Regex(BaseURL.UrlRegex, RegexOptions.Compiled);
        }

    }
}

File: Java/libraries/recognizers-text-sequence/src/main/java/com/microsoft/recognizers/text/sequence/english/extractors/EnglishURLExtractorConfiguration.java
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.

package com.microsoft.recognizers.text.sequence.english.extractors;

import com.microsoft.recognizers.text.sequence.SequenceOptions;
import com.microsoft.recognizers.text.sequence.config.URLConfiguration;
import com.microsoft.recognizers.text.sequence.resources.BaseURL;

import java.util.regex.Pattern;

public class EnglishURLExtractorConfiguration extends URLConfiguration {
    public EnglishURLExtractorConfiguration(SequenceOptions options) {
        super(options);

        this.setIpUrlRegex(Pattern.compile(BaseURL.IpUrlRegex));
        this.setUrlRegex(Pattern.compile(BaseURL.UrlRegex));
    }
}
